### PR TITLE
RSC: Differentiate routes autoloading behaviour

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**Problem**
We would like to load pages differently based on what rendering context we are in, the client browser or during SSR. 

**Changes**
1. Separates out the RSC route loading logic into its own plugin
2. Disables the non-RSC variant of the route loading plugin in the root vite config when RSCs are not enabled
3. Adds the RSC route loading plugin to both the client and server build steps with the appropriate client/server flag set

### CODE REVIEW

File: packages/babel-config/src/plugins/babel-plugin-redwood-rsc-routes-auto-loader.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import {
+  ensurePosixPath,
+  getPaths,
+  importStatementPath,
+  processPagesDir,
+} from '@redwoodjs/project-config'
+
+export interface PluginOptions {
+  forClient?: boolean
+  forServer?: boolean
```
Comment (Josh-Walker-GM) [inline packages/babel-config/src/plugins/babel-plugin-redwood-rsc-routes-auto-loader.ts:15]: Maybe this should be `forSSR`?

**Comment by Josh-Walker-GM** [comment]
This was merged into #10238 and will make it's way into main via that route.

File: packages/vite/src/rsc/rscWorker.ts [hunk 2]
```diff
@@ -121,6 +125,17 @@ registerFwGlobals()
 // is already in use`.
 const dummyServer = new Server()
 
+const reactBabelConfig = getWebSideDefaultBabelConfig({
+  forVite: true,
+  forRSC: true,
+})
+reactBabelConfig.overrides.push({
+  test: /Routes.(js|tsx|jsx)$/,
+  plugins: [[RedwoodRoutesAutoLoaderRscServerPlugin, {}]],
```
Comment (Tobbe) [inline packages/vite/src/rsc/rscWorker.ts:134]: You don't need an array of arrays here, right?
```suggestion
  plugins: [RedwoodRoutesAutoLoaderRscServerPlugin],
```

Comment (Josh-Walker-GM) [reply packages/vite/src/rsc/rscWorker.ts:134]: I'll double-check when the CI goes green. There was some issue with some of the jest tests failing for no reason when you didn't pass the empty object. Yeah, I'll go through the code when it passes CI to refactor and tidy.

File: packages/babel-config/src/plugins/babel-plugin-redwood-routes-auto-loader-rsc-client.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  importStatementPath,
+  processPagesDir,
+} from '@redwoodjs/project-config'
+
+import {
+  getPathRelativeToSrc,
+  withRelativeImports,
+} from './babel-plugin-redwood-routes-auto-loader'
+
+export function RedwoodRoutesAutoLoaderRscClientPlugin({
```
Comment (Tobbe) [inline packages/babel-config/src/plugins/babel-plugin-redwood-routes-auto-loader-rsc-client.ts:14]:
```suggestion
export function redwoodRoutesAutoLoaderRscClientPlugin({
```
I'd prefer this to start with a lowercase letter

File: packages/babel-config/src/plugins/babel-plugin-redwood-routes-auto-loader-rsc-server.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  importStatementPath,
+  processPagesDir,
+} from '@redwoodjs/project-config'
+
+import {
+  getPathRelativeToSrc,
+  withRelativeImports,
+} from './babel-plugin-redwood-routes-auto-loader'
+
+export function RedwoodRoutesAutoLoaderRscServerPlugin({
```
Comment (Tobbe) [inline packages/babel-config/src/plugins/babel-plugin-redwood-routes-auto-loader-rsc-server.ts:14]:
```suggestion
export function redwoodRoutesAutoLoaderRscServerPlugin({
```

**Pull request merged**